app/src/main/java/com/elyeproj/horizontaltextarragement/MainActivity.kt

```diff
@@ -30,5 +32,12 @@ class MainActivity : AppCompatActivity() {
                 space_worst_wrap.visibility = View.VISIBLE
             }
         }
+
+        btn_constraint_remove.setOnClickListener {
+            val isVisible = txt_worst_constraint.visibility == View.VISIBLE
+            TransitionManager.beginDelayedTransition(constraint_layout)
```

Comment:
This is not required, once we have `animateLayoutChanges="true"` in the ConstraitLayout.